packager/buildpack.go

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2018-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package packager
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/buildpacks/libcnb/v2"
+	"github.com/paketo-buildpacks/libpak/v2/carton"
+	"github.com/paketo-buildpacks/libpak/v2/effect"
+)
+
+type BundleBuildpack struct {
+	// BuildpackPath is the location to the buildpack source files
+	BuildpackPath string
+
+	// BuildpackID is the id of the buildpack you want to package
+	BuildpackID string
+
+	// Version is a version to substitute into an existing buildpack.toml.
+	BuildpackVersion string
+
+	// CacheLocation is the location to cache downloaded dependencies.
+	CacheLocation string
+
+	// DependencyFilters indicates which filters should be applied to exclude dependencies
+	DependencyFilters []string
+
+	// StrictDependencyFilters indicates that a filter must match both the ID and version, otherwise it must only match one of the two
+	StrictDependencyFilters bool
+
+	// IncludeDependencies indicates whether to include dependencies in build package.
+	IncludeDependencies bool
+
+	executor    effect.Executor
+	exitHandler libcnb.ExitHandler
+}
+
+func NewBundleBuildpack() BundleBuildpack {
+	return BundleBuildpack{
+		executor: effect.NewExecutor(),
+	}
+}
+
+func NewBundleBuildpackForTests(executor effect.Executor, exitHandler libcnb.ExitHandler) BundleBuildpack {
+	return BundleBuildpack{
+		executor:    executor,
+		exitHandler: exitHandler,
+	}
+}
+
+// InferBuildpackPath infers the buildpack path from the buildpack id
+func (p *BundleBuildpack) InferBuildpackPath() error {
```

Comment:
my compiler complains a bit cause you use a mix of pointer and value receiver methods on `BundleBuildpack`

WDYT?

Reply:
I don't get any warnings, but I think I know what you're talking about and I just pushed up a change to make them all pointers (cause we need pointer for one of the methods).